src/Jackalope/ObjectManager.php

```diff
@@ -779,9 +781,17 @@ protected function pathArrayToPropertiesIterator($propertyPaths)
      *
      * @param  string  $cnd         a string with cnd information
      * @param  boolean $allowUpdate whether to fail if node already exists or to update it
-     * @return bool    true on success
+     * @return bool|\Iterator    true on success or \Iterator over the registered node types if repository is not able to process
```

Comment:
i think this is not correct. this method always registers the node types, either directly through transport or by parsing the file and calling the node type manager registerNodeTypes method.

if neither is possible, it throws UnsupportedRepositoryOperationException

so no boolean is ever returned.